packages/suite/src/views/onboarding/steps/SecurityCheck/SecurityCheck.tsx

```diff
@@ -289,25 +300,47 @@ export const SecurityCheckContent = ({
 
 export const SecurityCheck = () => {
     const device = useSelector(selectDevice);
-
+    const devices = useSelector(selectDevices);
+    const deviceAuthenticity = useSelector(selectDeviceAuthenticity);
+    const dispatch = useDispatch();
     const [isDeviceAuthenticityCheck, setIsDeviceAuthenticityCheck] = useState(false);
-
-    const isAuthenticityCheckSupported = ![
-        DeviceModelInternal.T1B1,
-        DeviceModelInternal.T2T1,
-    ].includes(device?.features?.internal_model || DeviceModelInternal.T1B1);
+    const { goToSuite } = useOnboarding();
+
+    const shouldAuthenticateSelectedDevice =
+        !!device?.features?.internal_model &&
+        isAuthenticityCheckSupported(device.features.internal_model);
+
+    // If there are multiple devices connected, make sure to authenticate all those that support the check before continuing to Suite.
+    const goToSuiteOrNextDevice = () => {
+        const nextDeviceToAuthenticate = devices.find(
+            device =>
+                device.features?.internal_model &&
+                isAuthenticityCheckSupported(device?.features?.internal_model) &&
+                device.id &&
+                !deviceAuthenticity?.[device.id],
+        );
+
+        if (nextDeviceToAuthenticate) {
+            if (setIsDeviceAuthenticityCheck) {
+                setIsDeviceAuthenticityCheck(false);
+            }
+            dispatch(deviceActions.selectDevice(nextDeviceToAuthenticate));
+        } else {
+            goToSuite();
+        }
+    };
 
     // Edge case:
     // Devices A and B are connected, only device A supports authenticity check.
     // Device A disconnects while on the first screen of the check.
     useEffect(() => {
-        if (!isAuthenticityCheckSupported) {
+        if (isDeviceAuthenticityCheck && !shouldAuthenticateSelectedDevice) {
             setIsDeviceAuthenticityCheck(false);
         }
-    }, [isAuthenticityCheckSupported]);
+    }, [isDeviceAuthenticityCheck, shouldAuthenticateSelectedDevice]);
 
     if (isDeviceAuthenticityCheck) {
```

Comment:
I know the code was here before, but I was struggling to understand what does this mean. 

I suggest something like `should(Show|Display|Render)DeviceAuthenticityChek`